### PR TITLE
Memory and disconnection leaks

### PR DESCRIPTION
**Pull request by rex-remind101**

### CODE REVIEW

**Comment by rex-remind101** [comment]
@LeffelMania please review

File: Library/PTURLRequestOperation.m [hunk 1]
```diff
@@ -50,19 +82,22 @@ - (void)start
 
   [self setExecuting:YES];
 
+  self.sessionDelegate = [PTURLRequestOperationURLSessionDelegate new];
+  self.sessionDelegate.delegate = self;
   NSURLSessionConfiguration *sessionConfig = [NSURLSessionConfiguration defaultSessionConfiguration];
-  URLSession = [NSURLSession sessionWithConfiguration:sessionConfig delegate:self delegateQueue:nil];
-  NSURLSessionDataTask *task = [URLSession dataTaskWithRequest:URLRequest];
+  _URLSession = [NSURLSession sessionWithConfiguration:sessionConfig delegate:self.sessionDelegate delegateQueue:nil];
```
Comment (rex-remind101) [inline Library/PTURLRequestOperation.m:88]: `NSURLSession` holds a `strong` reference to the `delegate`. So I made a proxy so `self` and possibly the rest of pusher isn't captured.

File: Library/PTPusher.m [hunk 2]
```diff
@@ -375,7 +376,7 @@ - (void)pusherConnection:(PTPusherConnection *)connection didDisconnectWithCode:
 {
   NSError *error = nil;
 
-  if (errorCode > 0) {
+  if (errorCode > 0 && errorCode != SRStatusCodeNormal && errorCode != SRStatusCodeGoingAway) {
```
Comment (rex-remind101) [inline Library/PTPusher.m:379]: `SRWebSocket` will return at least these two codes on disconnect, Pusher wasn't respecting these codes and continued to try to reconnect anyway.

**Comment by LeffelMania** [comment]
Looks good to me 👍

**Pull request merged**